File: boj/brute_force/17618/17618.py
import sys
input = sys.stdin.readline
n = int(input())
num = int(n)
ans = 0
for num in range(1, n+1):
    sum = 0
    stringN = str(num)
    lenN = len(stringN)
    for j in range(lenN):
        sum += int(stringN[j])
    if(num % sum == 0): ans+=1
print(ans)

File: boj/brute_force/17618/17618.cpp
#include <bits/stdc++.h>
using namespace std;
int n, ans;
int main(){
    cin >> n;
    for(int i = 1; i <= n; i++){
        string stringN = to_string(i);
        int sum = 0;
        for(int j = 0; j < stringN.size(); j++){
            sum += stringN[j] - '0';
        }
        if(i % sum == 0) ans++;
    }
    cout << ans;
}